codec_record.go

```diff
@@ -77,6 +84,21 @@ func decoderOfStruct(cfg *frozenConfig, schema Schema, typ reflect2.Type) ValDec
 			continue
 		}
 
+		if field.action == FieldSetDefault {
+			if field.hasDef {
+				fields = append(fields, &structFieldDecoder{
+					field:   sf.Field,
+					decoder: createDefaultDecoder(cfg, field.Type(), field.def, sf.Field[len(sf.Field)-1].Type()),
+				})
+			} else {
```

Comment:
This `else` is not needed if you `continue` in the above `if` block.

Reply:
Yep, that’s a mistake